### ISSUE TITLE
postgresql 11.5 is supported by harbor?

### ISSUE THREAD

**Issue by cmpmohan** [goharbor/harbor#9031]
What can we help you?

We have POC setup harbor with postgresql 11.5.

Image push/pull is working fine and  PostgreSQL repository tables getting record according to pull/push images.
can we continue the POC with same version of 11.5 or will go with 9.6, 10.8 version? please suggest us harbor supported postgresql version?

harbor=# select * from repository;
repository_id | name | project_id | description | pull_count | star_count | creation_time | update_time
---------------+-----------------+------------+-------------+------------+------------+----------------------------+----------------------------
1 | library/python | 1 | | 0 | 0 | 2019-09-04 14:54:16.190055 | 2019-09-04 14:54:16.190055
2 | mohan/redis | 2 | | 0 | 0 | 2019-09-04 14:55:39.354839 | 2019-09-04 14:55:39.354839
3 | mohan/postgres | 2 | | 0 | 0 | 2019-09-04 15:04:24.693983 | 2019-

**Comment by wy65701436** [comment]
We haven't tried Pg 11.5 yet, the supported version is at [here](https://github.com/goharbor/harbor/blob/master/docs/installation_guide.md#harbor-components).

**Comment by jacekn** [comment]
It's also unclear what's the minimum PG version that is supported. Will 9.5.19 work?

**Comment by wy65701436** [comment]
We do only tried on PG 9.6.10 so far and no plan to support higher version. @xaleeks any comments?

**Comment by xaleeks** [comment]
@jacekn We post the pgsql version used in the installation / user guide.  We don't really check compatibility against all versions of pgsql, that's a bit of overkill so the posted version is the officially supported version.  I'm not sure what the new pgsql 10 offers but usually upgrades allow for better table partitioning and concurrency etc.  If you can share some background on this, we can look into upgrading it at some point but right now 9.6 is the only supported version

**Comment by jacekn** [comment]
@xaleeks thanks for the info. FYI I filed #9333 asking for extra documentation on the subject.

I'm not aware of any psql 10 (or 11 for that matter) functionality that would benefit the project.
I'm coming from a sysadmin background and people like me are often responsible for running databases. Very specific version requirements introduce operational difficulties when those databases are run outside of k8s.
Some example things that people like me need to consider:
* how to apply postgres security patches when using harbor provided docker image?
* company "standard OS" might be ubuntu 18.04 which ships with PG 10.10. How do I install PG 9.6? Who provides security patches for PG9.6 in that case?
* ubuntu 18.04 will eventually bump postgres from 10.10 to 10.11 to get latest security patches. Will that work with harbor? Do I need to hold security patches back?
* if a company pays for Ubuntu support then the vendor will likely insist we run official postgres package that comes with the OS
* a company might have internal "database as a service" that does not support 9.6.10 for some reason

Having said that I actually tried harbor on postgres 10.10 and so far it seems to be fine.

For you benefit - postgres feature matrix says that version 10 and 11 only obsolete one feature:
https://www.postgresql.org/about/featurematrix/

which I doubt harbor uses.
